OrbitCore/ElfFile.cpp

```diff
@@ -0,0 +1,95 @@
+#include "ElfFile.h"
+
+#include <vector>
+
+#include "PrintVar.h"
+#include "OrbitFunction.h"
+#include "Path.h"
+#include "absl/strings/str_format.h"
+#include "llvm/Object/ELFObjectFile.h"
+#include "llvm/Object/ObjectFile.h"
+
+std::unique_ptr<ElfFile> ElfFile::Create(const std::string& file_path) {
+  std::unique_ptr<ElfFile> result(new ElfFile(file_path));
+  if (!result->Load()) {
+    return nullptr;
+  }
+
+  return result;
+}
+
+bool ElfFile::Load() {
+  auto object_file_or_error =
+      llvm::object::ObjectFile::createObjectFile(file_path_);
+
+  if (!object_file_or_error) {
+    return false;
+  }
+
+  file_ = std::move(object_file_or_error.get());
+
+  auto object_file = file_.getBinary();
+  // Find the .text section
+  for (llvm::object::SectionRef section : object_file->sections()) {
+    llvm::StringRef name;
+    section.getName(name);
+    if (name.str() == ".text") {
+      text_section_ = std::make_unique<llvm::object::SectionRef>(section);
+      break;
+    }
+  }
+
+  return true;
+}
+
+bool ElfFile::IsAddressInTextSection(uint64_t address) const {
+  if (!text_section_) {
+    PRINT(".text section was not found\n");
+    return false;
+  }
+
+  uint64_t section_begin_address = text_section_->getAddress();
+  uint64_t section_end_address =
+      section_begin_address + text_section_->getSize();
+
+  return (address >= section_begin_address) && (address < section_end_address);
+}
+
+bool ElfFile::GetFunctions(Pdb* pdb, std::vector<Function>* functions) const {
+  auto object_file = file_.getBinary();
```

Comment:
Can we use the explicit type instead of auto, it will make the code clearer.

Reply:
done